testing/sharedtesthelpers/helpers.go

```diff
@@ -0,0 +1,236 @@
+			len(expectedInventories),
+			Form(len(expectedInventories), "package", "packages"),
+			len(actualInventories),
+		)
+	}
+
+	missingActualPackages := findMissingPackages(t, actualInventories, expectedInventories)
+	missingExpectedPackages := findMissingPackages(t, expectedInventories, actualInventories)
+
+	if len(missingActualPackages) != 0 {
+		for _, unexpectedPackage := range missingActualPackages {
+			t.Errorf("Did not expect %s", packageToString(unexpectedPackage))
+		}
+	}
+
+	if len(missingExpectedPackages) != 0 {
+		for _, unexpectedPackage := range missingExpectedPackages {
+			t.Errorf("Did not find   %s", packageToString(unexpectedPackage))
+		}
+	}
+}
+
+type ScanInputMockConfig struct {
+	Path string
+	// FakeScanRoot allows you to set a custom scanRoot, can be relative or absolute,
+	// and will be translated to an absolute path
+	FakeScanRoot string
+	FakeFileInfo *fakefs.FakeFileInfo
+}
+
+type ScanInputWrapper struct {
+	fileHandle *os.File
+	ScanInput  filesystem.ScanInput
+}
+
+func (siw ScanInputWrapper) Close() {
+	siw.fileHandle.Close()
+}
+
+// Generate FileInfoMock will either use the fake file information if fakeFileInfo is true,
+// otherwise try to run os.Stat on the path passed in and fail if the file does not exist
+func GenerateFileInfoMock(t *testing.T, config ScanInputMockConfig) fs.FileInfo {
+	t.Helper()
+
+	if config.FakeFileInfo != nil {
+		ret := *config.FakeFileInfo
+		ret.FileName = filepath.Base(config.Path)
+
+		return ret
+	}
+
+	fileInfo, err := os.Stat(config.Path)
+	// It is intended that sometimes the config points to a path that does not exist
+	// fileInfo will be nil in those cases
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		t.Fatalf("Can't stat test fixture '%s' because '%s'", config.Path, err)
+	}
+
+	return fileInfo
+}
+
+// GenerateScanInputMock will try to open the file locally, and fail if the file doesn't exist
+func GenerateScanInputMock(t *testing.T, config ScanInputMockConfig) ScanInputWrapper {
+	t.Helper()
+
+	var scanRoot string
+	if filepath.IsAbs(config.FakeScanRoot) {
+		scanRoot = config.FakeScanRoot
+	} else {
+		workingDir, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("Can't get working directory because '%s'", workingDir)
+		}
+		scanRoot = filepath.Join(workingDir, config.FakeScanRoot)
+	}
+
+	f, err := os.Open(filepath.Join(scanRoot, config.Path))
+	if err != nil {
+		t.Fatalf("Can't open test fixture '%s' because '%s'", config.Path, err)
+	}
+	info, err := f.Stat()
+	if err != nil {
+		t.Fatalf("Can't stat test fixture '%s' because '%s'", config.Path, err)
+	}
+
+	return ScanInputWrapper{
+		fileHandle: f,
+		ScanInput: filesystem.ScanInput{
+			FS:     os.DirFS(scanRoot).(scalibrfs.FS),
+			Path:   config.Path,
+			Root:   scanRoot,
+			Reader: f,
+			Info:   info,
+		},
+	}
+}
+
+// Form returns the singular or plural form that should be used based on the given count
+func Form(count int, singular, plural string) string {
```

Comment:
If this is just used by just one function then maybe this can be private or inlined

Reply:
Made this private.